### PR TITLE
Stop duplicated uniforms from being added

### PR DESCRIPTION
**Pull request by maximumpower55**
Will close: #300

### CODE REVIEW

File: src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java [hunk 1]
```diff
@@ -155,8 +160,14 @@ public OptionalInt location(String name, UniformType type) {
 				return OptionalInt.empty();
 			}
 
-			locations.put(id, name);
-			uniformNames.put(name, type);
+			if (!locations.containsKey(id) && !uniformNames.containsKey(name)) {
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java:163]: Did you mean uniformNames?

Comment (coderbot16) [inline src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java:163]: The only purpose of externally managed uniforms is just to suppress warnings for uniforms not managed through the uniform system - there isn't a pressing need to check them, but if we do, we should throw a hard error instead. Not sure if we should check if a normal uniform collides with an externally managed uniform.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

**Comment by coderbot16** [comment]
Huh, looks like IntelliJ's built in PR review thing messed up the comment locations. In any case the first comment targeted the first check & the second targeted the second check.

File: src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java (hunk 1 above)
Comment (maximumpower55) [reply src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java:163]: Yes

**Comment by maximumpower55** [comment]
Made requested changes

**Review by coderbot16 — APPROVED** [review]
Thanks!

**Pull request merged**